Game/MyProject/Game/ResourceManager/ResourceManager.hpp

```diff
@@ -0,0 +1,62 @@
+//
+//  ResourceManager.hpp
+//  RacingGame
+//
+//  Created by Tamas Both - (p) on 17/04/2019.
+//
+
+#pragma once
+
+#include <list>
+#include <map>
+#include <functional>
+
+#include <assimp/Importer.hpp>
+#include <assimp/scene.h>
+#include <assimp/postprocess.h>
+#include <IvImage.h>
+#include <IvRenderer.h>
+#include <IvTexture.h>
+
+#include "IvResourceManagerOGL.h"
+
+class ResourceManager {
+public:
+    // typedefs
+    typedef std::shared_ptr<IvTexture>              TexturePtr;
+    typedef std::shared_ptr<const aiScene>          ConstAiScenePtr;
+    typedef std::map<std::string, TexturePtr>       StringToTexture;
+    typedef std::map<std::string, ConstAiScenePtr>  StringToAiScene;
```

Comment:
Never use std::list and std::map becuase they make the code slow. Always use std::unordered_map and std::vector instead.
https://www.youtube.com/watch?v=rX0ItVEVjHc